src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -864,4 +862,27 @@ else if (hasConfigurationFile) {
             return result;
         }
     }
+
+    /**
+     * This class is needed for a workaround for the Jacoco limitations. A call to
+     * {@link System#exit(int)} will never return, so Jacoco coverage probe will be missing.
+     * Since it is not possible to exclude a method from the Jacoco coverage,
+     * the entire class is excluded.
+     */
+    private static class ExitHelper {
+
+        /**
+         * Terminates the program.
+         *
+         * @param code the return code
+         * @noinspection CallToSystemExit
+         */
+        private static void exit(int code) {
+            if (code != 0) {
+                System.exit(code);
```

Comment:
Can we keep IF condition out of Helper class ?

As we keep this class only to make Jacoo happy 
So better to keep only problematic  statement.

Reply:
No. In this case the coverage of the `Main` class will be 99% because the `ExitHelper.exit()` will never return:

```java
 if (statusCode != 0) {
  ExiitHelper.exit(statusCode);
  // This line will be marked as uncovered
}
```

Reply:
you mean coverage will drop if we do:
```
        private static void exit(int code) {
                System.exit(code);
        } 
```

if it drops it means problem is in IF condition coverage.

what do you mean by `will never return` ? it will never finish to execute ? or it will not actually exit from java process ?

Reply:
```
✔ ~/java/github/romani/checkstyle [pbludov/issue-10932-main L|✚ 1] 
05:55 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
index 8d5a261..ae0724d 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
@@ -149,7 +149,9 @@ public final class Main {
                 System.err.println(errorCounterViolation.getViolation());
             }
         }
-        ExitHelper.exit(exitStatus);
+        if (exitStatus != 0) {
+            ExitHelper.exit(exitStatus);
+        }
     }
 
     /**
@@ -878,9 +880,7 @@ public final class Main {
          * @noinspection CallToSystemExit
          */
         private static void exit(int code) {
-            if (code != 0) {
-                System.exit(code);
-            }
+            System.exit(code);
         }
 
     }
```

![image](https://user-images.githubusercontent.com/812984/143880781-df5dd52b-1bc4-44d2-b68e-34b4370c66ab.png)